internal/datanode/channel_meta.go

```diff
@@ -294,13 +311,39 @@ func (c *ChannelMeta) filterSegments(partitionID UniqueID) []*Segment {
 }
 
 func (c *ChannelMeta) InitPKstats(ctx context.Context, s *Segment, statsBinlogs []*datapb.FieldBinlog, ts Timestamp) error {
+	if Params.DataNodeCfg.SkipBFStatsLoad {
+		// mark segment lazy loading
+		s.setLoadingLazy(true)
+		go func() {
+			c.workerPool.Submit(func() (any, error) {
+				stats, err := c.loadStats(context.Background(), s.segmentID, s.collectionID, statsBinlogs, ts)
+				if err != nil {
+					log.Warn("failed to lazy load statslog for segment", zap.Int64("segmentID", s.segmentID), zap.Int64("collectionID", s.collectionID))
+					return nil, err
```

Comment:
and also what happens if lazy load failed?
I think we should retry for a very long time. until some time we will just abort the service or release the channle

Reply:
But what it an io error or a S3 rate limit?

Reply:
Added re-submit logic and error type check logic here